gssutils/scrapers/govscot.py

```diff
@@ -9,26 +11,65 @@
-    for row in doctable.xpath('tr'):
-        try:
-            if row.xpath('th/text()')[0].strip() == 'File:':
-                cell = row.xpath('td')[0]
-                dist = Distribution(scraper)
-                title_size_date = re.compile(r'(.*)\[([^,]+),\s+([0-9.]+)\s+([^:]+):\s([^\]]+)')
-                match = title_size_date.match(cell.text_content())
-                if match:
-                    dist.title = match.group(1)
-                    scraper.dataset.issued = parse(match.group(5), dayfirst=True).date()
-                dist.downloadURL = urljoin(scraper.uri, cell.xpath('a/@href')[0])
-                dist.mediaType, encoding = mimetypes.guess_type(dist.downloadURL)
-                scraper.distributions.append(dist)
-        except:
-            break
+
+	def pubs(soup):
+
+		for a in soup.select('#page-content > div.top-matter > div > header > div:nth-child(1) > div > h1'):
+			scraper.dataset.title = a.text
+
+		for a in soup.select('#page-content > div.top-matter > div > header > div:nth-child(2) > div.grid__item.large--seven-twelfths > div > p'):
+			scraper.dataset.description = a.text
+
+		for a in soup.select('#page-content > div.inner-shadow-top.js-sticky-header-position.js-subpage-top-edge > div > div > div.grid__item.large--seven-twelfths > section'):
+			a.xpath('div:nth-child(1) > div.document-info__text > h3 > a')
+
+		try:
+			pubDate = soup.findAll('span', class_= 'content-data__value')[0].text
+			scraper.dataset.issued = pubDate
+		except:
+			pass
+
+		dists = soup.findAll(class_= 'document-info')
+
+		for entry in dists:
+			distributions = entry.findAll(class_= 'no-icon')
+			for i in distributions:
+				dist = Distribution(scraper)
+				dist.title = i.text
+				dist.downloadURL = urljoin("https://www.gov.scot", i['href'])
+				dist.mediaType, encoding = mimetypes.guess_type(dist.downloadURL)
+				if dist.mediaType in ['application/msword', 'application/pdf', 'None']:
+					return
+				elif dist.mediaType == None:
+					return
+				else:
+					scraper.distributions.append(dist)
+
+	def colls(soup):
+
+		pubPages = soup.findAll('ul', class_='collections-list')
+		for page in pubPages:
+			pages = page.findAll('a')
+			for i in pages:
+				req = Request(urljoin("https://www.gov.scot", i['href']), headers={'User-Agent': 'Mozilla/5.0'})
+				html = urlopen(req).read()
+				plaintext = html.decode('utf8')
+				soup = BeautifulSoup(plaintext, 'html.parser')
+				pubs(soup)
+
+	r = scraper.session.get(scraper.uri)
+
+	parse_object = urlparse(r.url)
+
+	soup = BeautifulSoup(r.text, features="lxml")
+
+	scraper.dataset.publisher = GOV['the-scottish-government']
+
+	scraper.dataset.license = 'http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/'
+
+	if 'publications' in parse_object.path:
+		pubs(soup)
+	elif 'collections' in parse_object.path:
+		colls(soup)
+	else:
+		print('GovScot Scraper does not support given landing page')
```

Comment:
```suggestion
		print('GovScot Scraper does not support given landing page')

```
Generally it's a good idea to have a blank line at the end since some tools misbehave when it's not there